sql/core/src/main/scala/org/apache/spark/sql/execution/command/DataWritingCommand.scala

```diff
@@ -96,4 +98,23 @@ object DataWritingCommand {
       sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY),
       metrics.values.toSeq)
   }
+
+  def assertEmptyRootPath(tablePath: URI, saveMode: SaveMode, sparkSession: SparkSession) {
+    if (saveMode != SaveMode.Overwrite &&
+      !sparkSession.sqlContext.conf.allowCreateTableAsSelectNonEmptyLocation) {
+      val filePath = new org.apache.hadoop.fs.Path(tablePath)
+      val fs = filePath.getFileSystem(sparkSession.sparkContext.hadoopConfiguration)
+      if(fs != null && fs.exists(filePath)) {
+        val locStats = fs.getFileStatus(filePath)
+        if(locStats != null && locStats.isDirectory) {
+          val lStats = fs.listStatus(filePath)
+          if(lStats != null && lStats.length != 0) {
+            throw new AnalysisException(
+              s"CREATE-TABLE-AS-SELECT cannot create table" +
```

Comment:
I would say something like this:
> The directory ${tablePath} was not empty but CTAS with LOCATION requires the location to be empty.

See also https://spark.apache.org/error-message-guidelines.html

Reply:
We should also probably mention the legacy configuration here.

Reply:
In Hive , https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/ql/ErrorMsg.java#L408
Error message is `CREATE-TABLE-AS-SELECT cannot create table with location to a non-empty directory.`
To be consistent, shall we show this message?
 "CREATE-TABLE-AS-SELECT cannot create table with location to a non-empty directory.  To disable the behavior, set 'spark.sql.legacy.allowNonEmptyLocationInCTAS to true"